### ISSUE TITLE
Quickstart iOS test fails

### ISSUE THREAD

**Issue by artemyarulin** [facebook/buck#610]
I've just created an ios test app using quickstart and it fails to run the tests:
``` bash
> buck quickstart --type ios --dest-dir app-ios
> cd app-ios
> buck build //ios:BuckDemoApp
> buck test demo_app_ios
ios/tests/BuckDemoAppTest.m:18:9: fatal error: 'AppDelegate.h' file not found
#import "AppDelegate.h"
        ^
1 error generated.

ios/AppViewController.m:21:17: warning: method override for the designated initializer of the superclass '-initWithNibName:bundle:' not found [-Wobjc-designated-initializers]
@implementation AppViewController {
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:104:1: note: method marked as designated initializer of the class here
- (instancetype)initWithNibName:(nullable NSString *)nibNameOrNil bundle:(nullable NSBundle *)nibBundleOrNil NS_DESIGNATED_INITIALIZER;
^
ios/AppViewController.m:21:17: warning: method override for the designated initializer of the superclass '-initWithCoder:' not found [-Wobjc-designated-initializers]
@implementation AppViewController {
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:105:1: note: method marked as designated initializer of the class here
- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
^
2 warnings generated.

BUILD FAILED: //ios:BuckDemoAppTest#apple-test-library,compile-pic-tests_BuckDemoAppTest.m.o,iphonesimulator-x86_64 failed with exit code 1:
objective-c compile_munge_debuginfo
[-] PROCESSING BUCK FILES...FINISHED 0.0s
[-] BUILDING...FINISHED 0.5s [100%] (11/22 JOBS, 6 UPDATED, 27.3% CACHE MISS)
```

I'm very new to the buck and don't know where to start looking for the cause for an issue. I'm using buck version v2015.12.14.01, with latest XCode/MacOS

Feel free to ask more question or maybe you can guide me where I should start looking for the cause of an issue

UPD: Updated xctool till 0.2.7, no more warning but the error is the same. While build and install works just fine

**Comment by LegNeato** [comment]
More info:

https://developer.apple.com/library/ios/documentation/General/Conceptual/CocoaEncyclopedia/Initialization/Initialization.html#//apple_ref/doc/uid/TP40010810-CH6-SW3

**Comment by ryu2** [comment]
The warnings are something we'll fix on our end, but the missing `AppDelegate.h` is strange.  Does it actually exist in the `ios` subdirectory?

**Comment by artemyarulin** [comment]
Yep, as I mentioned I can build and even run it without any problems. Here project tree:
```
> tree .
.
├── AppDelegate.h
├── AppDelegate.m
├── AppViewController.h
├── AppViewController.m
├── BUCK
├── Default-568h.png
├── Default-667h.png
├── Default-736h.png
├── Icon-60.png
├── Icon-72.png
├── Icon-72@2x.png
├── Icon-Small-50.png
├── Icon-Small-50@2x.png
├── Icon.png
├── Icon@2x.png
├── Info.plist
├── iTunesArtwork.png
├── iTunesArtwork@2x.png
├── main.m
└── tests
    ├── BuckDemoAppTest.m
    └── Test.plist

1 directory, 21 files
```
And here goes my BUCK file:
```
>cat BUCK 
apple_resource(
  name = 'BuckDemoAppResources',
  files = glob(['*.png']),
  dirs = [],
)

apple_bundle(
  name = 'BuckDemoApp',
  binary = ':BuckDemoAppBinary',
  extension = 'app',
  info_plist = 'Info.plist',
  tests = [':BuckDemoAppTest'],
)

apple_binary(
  name = 'BuckDemoAppBinary',
  deps = [':BuckDemoAppResources'],
  preprocessor_flags = ['-fobjc-arc'],
  headers = glob([
    '*.h',
  ]),
  srcs = glob([
    '*.m',
  ]),
  frameworks = [
    '$SDKROOT/System/Library/Frameworks/UIKit.framework',
    '$SDKROOT/System/Library/Frameworks/Foundation.framework',
  ],
)

apple_package(
  name = 'BuckDemoAppPackage',
  bundle = ':BuckDemoApp',
)

apple_test(
  name = 'BuckDemoAppTest',
  extension = 'xctest',
  test_host_app = ':BuckDemoApp',
  srcs = ['tests/BuckDemoAppTest.m'],
  info_plist = 'tests/Test.plist',
  frameworks = [
    '$SDKROOT/System/Library/Frameworks/Foundation.framework',
    '$PLATFORM_DIR/Developer/Library/Frameworks/XCTest.framework',
    '$SDKROOT/System/Library/Frameworks/UIKit.framework',
  ],
)
```

So, it looks like nobody can reproduce that? I'll try then on a different Mac then, maybe something wrong with my env then

**Comment by mremick** [comment]
I'm having the same issue, but not the app delegate issue.

**Comment by ryu2** [comment]
The warnings were fixed in 53bc7fa23fe2217db2e241cd914546a6001e53fd.  What other issues are you referring to?

**Comment by nayrmi** [comment]
Closing out since we have heard from folks in a few months. However, @mremick - feel free to reactivate if you have more info to provide!
